n_queen_problem/NQueenProblem.java

```diff
@@ -0,0 +1,53 @@
+import java.util.*;
+
+public class NQueenProblem { 
+    private static void placeQueenOnBoard(int Qi, int[] board) {
+        int n = board.length;
+        //base case
+        if (Qi == n) {
+            System.out.println(Arrays.toString(board));
```

Comment:
`board` in line 8 and after execution of line 51 in `main` would be same. We do not need to modify much for printing outside the algorithm function.